app/src/main/java/dracula/vlad/androcalc/MainActivity.java

```diff
@@ -221,65 +223,62 @@ public void getChildViews(ViewGroup viewGroup) {
             int children = viewGroup.getChildCount();
             for (int i = 0; i < children; i++) {
                 View view = viewGroup.getChildAt(i);
+
                 if (view instanceof ViewGroup) {
                     getChildViews((ViewGroup) view);
                 }
+
                 if (view instanceof Button) {
                     TypedValue outValue = new TypedValue();
                     view.getContext().getTheme().resolveAttribute(android.R.attr.selectableItemBackground,
                             outValue, true);
                     view.setBackgroundResource(outValue.resourceId);
-                    buttonArrayList.add((Button) view);
+                    buttons.add((Button) view);
                 }
             }
         }
     }
 
-    private void delAtCursorPos(){
-        final EditText editText = findViewById(R.id.et_Input);
-        int index = editText.getSelectionStart();
-        if(index!=0)
-            editText.getText().delete(index - 1, index);
-    }
-
-    private void addAtCursorPos(String curBtnText){
-        final EditText editText = findViewById(R.id.et_Input);
+    private void delAtCursorPos() {
+        int index = inputView.getSelectionStart();
 
-        int lengthET = editText.getText().length();
-        int index = editText.getSelectionStart();
-        Log.d(TAG, "index: "+index);
-
-        String full = editText.getText().toString();
+        if (index != 0) {
+            inputView.getText().delete(index - 1, index);
+        }
+    }
 
-        String first = full.substring(0,index);
-        Log.d(TAG, "first: "+first);
+    private void addAtCursorPos(String currentButtonText) {
+        int inputLength = inputView.getText().length();
+        int index = inputView.getSelectionStart();
+        Log.d(TAG, "index: " + index);
 
-        String second = full.substring(index,lengthET);
-        Log.d(TAG, "second: "+second);
+        String fullString = inputView.getText().toString();
+        Log.d(TAG, "full string: " + fullString);
 
-        editText.setText(String.format("%s%s%s", first, curBtnText, second));
+        String beforeCursorText = fullString.substring(0, index);
+        Log.d(TAG, "first: " + beforeCursorText);
 
-        Log.d(TAG, "lengthET: "+lengthET);
+        String afterCursorText = fullString.substring(index, inputLength);
+        Log.d(TAG, "second: " + afterCursorText);
 
-        if (lengthET>0)
-            editText.setSelection(index + 1);
-        else editText.setSelection(1);
+        inputView.setText(String.format("%s%s%s", beforeCursorText, currentButtonText, afterCursorText));
+        inputView.setSelection(inputLength > 0 ? index + 1 : 1);
     }
 
-    private void openDialog(){
+    private void openDialog() {
         AlertDialog.Builder builder = new AlertDialog.Builder(this);
         builder.setMessage("Скопировать результат в буфер?")
                 .setCancelable(true)
                 .setPositiveButton("Да", (dialog, id) -> {
```

Comment:
string resource with posibility of translation! also there are predefined system strings like android.R.string.yes which has translations in the system level and don't need to have them in your project